packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -262,27 +262,31 @@ abstract class ImageProvider<T> {
     assert(configuration != null);
     final ImageStream stream = ImageStream();
     T obtainedKey;
+    Future<void> handleError(dynamic exception, StackTrace stack) async {
+      await null; // wait an event turn in case a listener has been added to the image stream.
+      final _ErrorImageCompleter imageCompleter = _ErrorImageCompleter();
+      stream.setCompleter(imageCompleter);
+      imageCompleter.reportError( // ignore: invalid_use_of_protected_member
```

Comment:
The ignore seems fishy. Should the private subclass _ErrorImageCompleter just expose some form of reportError as public? Or could all these things be passed into the constructor and it internally calls reportError or something like that?

Reply:
Yes, I could just add a new method to the _ErrorImageCompleter - will do that instead